discovery/snowflake/spcs_test.go

```diff
@@ -0,0 +1,379 @@
+						 `,
+			expected: SPCSSDConfig{},
+			errMsg:   "Spcs SD configuration requires an Authenticator to be specified. Supported Authenticator types are AuthTypeToken, AuthTypePassword",
+		},
+		{
+			name: "Unsupported Authenticator",
+			yamlData: `
+						 account: "testOrganization-testAccount"
+						 authenticator: "InvalidAuthenticator"
+						 `,
+			expected: SPCSSDConfig{},
+			errMsg:   "Spcs SD configuration unsupported Authenticator InvalidAuthenticator. Supported Authenticator types are AuthTypeToken, AuthTypePassword",
+		},
+		{
+			name: "Missing User, Password, and Role for AuthTypePassword",
+			yamlData: `
+						 account: "myAccount"
+						 host: "myHost"
+						 user: "myUser"
+						 authenticator: "AuthTypePassword"						
+						 `,
+			expected: SPCSSDConfig{},
+			errMsg:   "Spcs SD configuration requires { User, Password, Role } when Authenticator type is AuthTypePassword",
+		},
+	}
+
+	unmarshal := func(d []byte) func(interface{}) error {
+		return func(o interface{}) error {
+			return yaml.Unmarshal(d, o)
+		}
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var config SPCSSDConfig
+			err := config.UnmarshalYAML(unmarshal([]byte(removeLeadingSpaces(test.yamlData))))
+			if test.errMsg != "" {
+				require.NotNil(t, err, "Expected an error")
+				require.Equal(t, test.errMsg, err.Error(), "Error message mismatch")
+			} else {
+				require.Nil(t, err, "Expected no error")
+				require.Equal(t, test.expected, config, "Config mismatch")
+			}
+		})
+	}
+}
+
+// Implement a mock resolver for testing.
+type MockEndpointResolver struct {
+	ipAddresses []string
+}
+
+func (m *MockEndpointResolver) ResolveEndPoint(endPoint string) ([]string, error) {
+	// Return predefined IP addresses for testing.
+	return m.ipAddresses, nil
+}
+
+func removeLeadingSpaces(input string) string {
+	lines := strings.Split(input, "\n")
+	for i, line := range lines {
+		lines[i] = strings.TrimLeft(line, " \t")
+	}
+	return strings.Join(lines, "\n")
+}
+
+func verifyResult(t *testing.T, err error, result, expected []string) {
+	if err != nil {
+		t.Errorf("Expected no error, but got an error: %v", err)
+	}
+	if len(result) != len(expected) {
+		t.Errorf("Expected %v, but got %v", expected, result)
+	}
+	for i := range expected {
+		if result[i] != expected[i] {
+			t.Errorf("Expected %v, but got %v", expected[i], result[i])
+		}
+	}
+}
+
+func verifyTargets(t *testing.T, err error, resultTargets, expectedTargets []model.LabelSet) {
+	if err != nil {
+		t.Errorf("Expected no error, but got: %v", err)
+	}
+
+	if !labelSetsEqual(resultTargets, expectedTargets) {
+		t.Errorf("Expected %v, but got %v", expectedTargets, resultTargets)
+	}
+}
+
+func labelSetsEqual(a, b []model.LabelSet) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if !labelSetEqual(a[i], b[i]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func labelSetEqual(a, b model.LabelSet) bool {
```

Comment:
Can you use this instead?
https://github.com/prometheus/common/blob/8e5d54a99b1dfaf2a5e7b076667257fc1d72a1c3/model/labelset.go#L44-L45